normalized-cache/src/commonMain/kotlin/com/apollographql/cache/normalized/internal/DefaultCacheManager.kt

```diff
@@ -164,7 +167,27 @@ internal class DefaultCacheManager(
       val jsonReader = mapOf("data" to it).jsonReader()
       jsonReader.beginObject()
       jsonReader.nextName()
-      val data = operation.adapter().fromJson(jsonReader, falseVariablesCustomScalarAdapter)
+      val data = runCatching {
+        operation.adapter().fromJson(jsonReader, falseVariablesCustomScalarAdapter)
+      }.getOrElse { e ->
```

Comment:
How doable is it to have partial data here? If only a field has an unexpected shape, only cache miss that one?